pkg/minikube/cluster/commands_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	gflag "flag"
+	"fmt"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"strings"
+	"testing"
+)
+
+func TestGetStartCommandDefaultValues(t *testing.T) {
+	flagMap := map[string]string{
+		"logtostderr":      "false",
+		"alsologtostderr":  "false",
+		"stderrthreshold":  "2",
+		"log_dir":          "",
+		"log_backtrace_at": ":0",
+		"v":                "0",
+		"vmodule":          "",
+	}
+	expectedStartCommand := genStartCommandFromFlagMap(flagMap)
+	flagMapToSetFlags(flagMap)
+	if startCommand := GetStartCommand(); expectedStartCommand != startCommand {
+		t.Fatalf("Expected FlagMapToFlagString to return: %s. Instead it was: %s", expectedStartCommand, startCommand)
+	}
+}
+
+func TestGetStartCommandCustomValues(t *testing.T) {
+	flagMap := map[string]string{
+		"logtostderr":      "true",
+		"alsologtostderr":  "true",
+		"stderrthreshold":  "3",
+		"log_dir":          "/var/",
+		"log_backtrace_at": "cluster.go:123",
+		"v":                "10",
+		"vmodule":          "cluster*=5",
+	}
+	expectedStartCommand := genStartCommandFromFlagMap(flagMap)
+	flagMapToSetFlags(flagMap)
+	if startCommand := GetStartCommand(); expectedStartCommand != startCommand {
+		t.Fatalf("Expected FlagMapToFlagString to return: %s. Instead it was: %s", expectedStartCommand, startCommand)
+	}
+}
+
+func flagMapToSetFlags(flagMap map[string]string) {
+	for k, v := range flagMap {
+		gflag.Set(k, v)
+	}
+}
+
+func genStartCommandFromFlagMap(flagMap map[string]string) string {
```

Comment:
This function looks basically the same as the implementation above, which means the test will be a little brittle. What about just asserting that each flag and it's value is present in the list (since we don't necessarily care about order). Something like

getSingleFlagValue(flag, value string) string {
  return fmt.Sprintf("--%s %s", flag, value)
}

Then your test loop over everyting in flagMap and assert that the individual flag is present with strings.Contains